### ISSUE TITLE
Feature Request: Use Snap channels to separate stable/beta builds

### ISSUE THREAD

**Issue by nmd3** [wekan/wekan#1966]
Would it be possible to begin using channels for snap releases as documented here: https://forum.snapcraft.io/t/channels/551

Currently Wekan channels are all the same release, and often there are multiple simultaneous releases to the point where I don't have the "latest" version despite using snap:
![image](https://user-images.githubusercontent.com/44180132/47433474-8dc9ff00-d76e-11e8-88e9-26c701a1701e.png)

What is actually more concerning to me is introducing new features to my users that might require a learning curve. It would be great to see most of the daily updates occurring on a candidate or beta channel and then added to stable releases on a schedule that I can anticipate (e.g. ~1/month). 

I love all of the work that you're doing, I'm just worried that introduction/removal of new features so rapidly is going to confuse my users and potentially break things (e.g. I've instructed them not to use custom fields because I don't want to repair JSON files every time they decide to delete the custom fields; I've also told them to avoid using subtasks due to the way the feature behaves). The latest version reintroduced rules, which canbe a great feature, but might be one that requires a learning curve.

Maybe an alternative to this is admin ability to disable certain features.

**Comment by xet7** [comment]
Maybe your updates are stuck.

1) Make backup
https://github.com/wekan/wekan/wiki/Backup

2) Change to stable channel.
```
sudo snap refresh wekan --stable
```

IFTTT Rules is only visible to board admin users. Non-admin users don't see IFTTT Rules at all. IFTTT, LDAP etc was for a long time on edge, and because there was no major bugs, edge was merged to stable.

I am not aware any latest changes have broken anything. There have been bugfixes for sending email etc.

Many do use buggy features like custom fields and subtasks. Removing them would cause breakage. It's better to fix those features sometime.

I could think about updating stable channel only once a month, but that would also mean that bugs that are fixed on edge would apprear to stable only once a month, and many would complain about those bugs that are already fixed on edge.

There is not very many config / Admin Panel options to switch features off yet. If someone is interested in that, please add feature request or PR.

**Comment by nmd3** [comment]
Updates are not stuck - you're working quickly enough that my overnight update only caught up to 1.58 rather than the releases a few hours later. :)

I was just suggesting that newer features be released to the candidate/beta/edge channels before merging into stable, but I understand that it might change your workflow in a manner that cannot be managed right now. Since there aren't release notes, I may end up switching to a docker deployment to more closely control the introduction of new features to my users, but that remains to be seen.

**Comment by xet7** [comment]
@nmd3 

There is release notes at https://wekan.github.io/ at buttons `Stable ChangeLog` and `Edge ChangeLog`. There are all changes to Wekan ever made.

**Comment by nmd3** [comment]
> There is release notes at https://wekan.github.io/ at buttons `Stable ChangeLog` and `Edge ChangeLog`. There are all changes to Wekan ever made.

Thanks - still learning my way around apparently

**Comment by xet7** [comment]
@nmd3 

At Stable Download page Platforms:
https://github.com/wekan/wekan/wiki/Platforms

There is described that Snap is for automatic updates, and Docker is if you test first before update.

**Comment by xet7** [comment]
@nmd3 

It's also possible to schedule Wekan Snap to update only at night, that way it does not update during workday:
https://github.com/wekan/wekan-snap/wiki/Install#5-install-all-snap-updates-automatically-between-0200am-and-0400am

**Comment by nmd3** [comment]
> 
> 
> @nmd3
> 
> It's also possible to schedule Wekan Snap to update only at night, that way it does not update during workday:
> https://github.com/wekan/wekan-snap/wiki/Install#5-install-all-snap-updates-automatically-between-0200am-and-0400am

Yes - this is exactly how I have my setup configured right now. I will probably consider switching to Docker at some point, especially if we have less sophisticated users on our setup. If there's suddenly a new feature available to them that I did not examine, they may come to me with questions before I've had the chance to familiarize myself with the features. 

If you don't think it's feasible to further decompose into separate snap channels, feel free to close. I appreciate the feedback on the intent for using docker for the scenario described.

**Comment by xet7** [comment]
I had separate channels of stable and edge before. In that case, I moved broken features from stable to edge. Edge had very large features like IFTTT and LDAP, and when they came stable enough, I merged edge to stable, because of a lot of requests for it. That also means that bugs that were not noticed at edge, get bug reports when they came to stable.

I still do small test at edge before adding the same package to stable.

I don't know yet what I should move to from stable to edge. Many Wekan features have a lot of code all around Wekan codebase, so it's more complicated than if each feature was in one separate file.

Would it work if I move from Stable to Edge:
- Custom Fields
- Subtasks

And when they are fixed, move back to Stable. Other option is to fix those features, and not spend time moving those features around.

I have also had comments about subtasks, that because they go to different subtask board, it makes many extra boards, so Subtasks feature should be removed or disabled. When someone moved cards away from subtask board, it broke the main board.

Wekan is community driven project. Sometimes some UI change is added, and there is feedback that original is better, so I change back to original. Also at #1429 some seem to want that Archive, that was renamed to Recycle Bin, should be renamed back to Archive. So maybe that should also be done someday.

In Summary, I don't know yet. Comments welcome.

**Comment by xet7** [comment]
@nmd3 

> I will probably consider switching to Docker at some point, especially if we have less sophisticated users on our setup. If there's suddenly a new feature available to them that I did not examine, they may come to me with questions before I've had the chance to familiarize myself with the features.

What would Wekan Snap Stable version look like?

What if current Wekan is the Stable version:
- Receives bugfixes
- Buggy and complex features will be changed to be disabled, and be as optional settings to Admin Panel and Snap settings.
- New UI changes will be optional, to be enabled separately at Admin Panel and Snap Settings

What do you think?

**Comment by nmd3** [comment]
> @nmd3
> 
> > I will probably consider switching to Docker at some point, especially if we have less sophisticated users on our setup. If there's suddenly a new feature available to them that I did not examine, they may come to me with questions before I've had the chance to familiarize myself with the features.
> 
> What would Wekan Snap Stable version look like?
> 
> What if current Wekan is the Stable version:
> 
> * Receives bugfixes
> * Buggy and complex features will be changed to be disabled, and be as optional settings to Admin Panel and Snap settings.
> * New UI changes will be optional, to be enabled separately at Admin Panel and Snap Settings
> 

I think your suggestion sounds good! 

I would consider not disabling _existing_ buggy/complex features by default in case it breaks the boards when people auto-update. For example, one of my users right now is using subtasks, but I have suggested not using them to him. If you were to make a change to the stable channel, I would want to make sure that it did not break his current usage for the board that he owns - though I am discouraging him from using them since they are currently filling up the home screen of other users...

I think having an opt-in strategy for enabling "beta" features might be a great idea - thinking something along the lines of what "Labs" was like in Gmail. 

The opt-in strategy would allow admins to enable the buggier features at their own risk. (E.g. I would probably disable custom fields and subtasks for my install.) 

Edit: typo... "I would consider not disabling _existing_ buggy/complex features"
